Add `zookeeper` action which leverages found-shell tree exported to include ZK (ciphered) contents in the output bundle.

pfcoperez opened the pull request:
Adding three new options to the diagnosis bundle generator:

- `zk|--zookeeper` Enables Zookeeper contents dump, it requires a path to a PGP public key (which Elastic engineers will provide and which its private counterpart is owned only by then). 
- `zk-path|--zookeeper-path` By default, the ZK contents dump is starts at the root of the repository (`/`). This option changes that behaviour allowing faster dumps by selecting paths support engineers might be interested in. 
- `zk-excluded|--zookeeper-excluded` Also, the default behaviour is to include all paths under the starting point (see point above). With this option, that behaviour can be changed as a, comma separated, list of concrete paths to avoid can be included. e.g: `/locks,/secrets/,/services/allocators` will remove these paths from the resulting bundle.

When `zk|--zookeeper` is present, the bundle will include `/elastic/zookeeper_dump/zkdump.zip.gpg` fie. It is encrypted using PGP and the key provided as a non-optional parameter. Only owners of the private key can open this file.

**Example of usage**

```bash
~/diagnostics.sh -zk ~/destination.key -zk-path /services -zk-excluded /services/allocators,/services/blueprint/states -d
```

**Internal**

This feature will only work with releases including this change: https://github.com/elastic/cloud/pull/42339

**This PR is ready to merge, I am holding it back until ECE release includes the infrastructure it requires (ECE >2.4)**

s-nel commented on diagnostics.sh:
```diff
@@ -196,6 +206,69 @@ get_docker(){
+                        excluded_nodes=","
+        fi
+
+        #Collect result at $zookeeper_folder/zkdump.zip
```
Should we put the cleartext in `/tmp` instead to make sure it doesn't get accidentally included?

pfcoperez replied:
I thought about that but the ZK bundle can grow quite fast. In my Fedora 29 set-up `/tmp` is mounted with the following options:

```
rw,nosuid,nodev,noexec,relatime,seclabel,size=65536k
```

That is, it is limited to 64 MiB, most bundles generations would fail if we used `/tmp` for this :cry:

pfcoperez replied:
Addressed in a2b04bd055e164513e966d128d56332cb1dee584 by taking the sibling directory to the one that contains the tree with the whole bundle. Then, the encrypted zk bundle gets included into this tree and the temporary sibling gets removed.

s-nel commented:
Should we update the README at all?

gcmurphy commented on diagnostics.sh:
```diff
@@ -6,6 +6,7 @@ diag_folder=$output_path/$diag_name
 elastic_folder=$diag_folder/elastic
 docker_folder=$diag_folder/docker
 docker_logs_folder=$docker_folder/logs
+zookeeper_folder=$elastic_folder/zookeeper_dump
```
Being picky here but could we use something like this instead?

```sh
zookeeper_folder=`mktemp -d $elastic_folder/zkdump.XXXXXX`
```

pfcoperez replied:
Addressed in a2b04bd055e164513e966d128d56332cb1dee584

gcmurphy commented on diagnostics.sh:
```diff
@@ -196,6 +206,69 @@ get_docker(){
+
+        mkdir $temp_keyring
+
+        gpg2 --homedir $temp_keyring --import $public_key_file
```
I think gpg is probably a good option here. The only thought I had was the scenario where the contents of the dump need to be passed between multiple support engineers during change of shift. Might need to think about a SDH process around that?

gcmurphy replied:
Hmm so maybe something like this?

```bash
#!/bin/bash 

# File to encrypt using AES256
ZIPFILE=#<The file name>

# On call support engineer ID
RECIPIENT=#<GPG key ID>

# Password generated or supplied by user
PW=$(uuid)

# File containing encrypted password. Encrypted by support engineers gpg key.
PWFILE=support.gpg

openssl enc -aes256 -e -in $ZIPFILE -out $ZIPFILE.enc -pass "pass:$PW" && echo $PW | gpg --encrypt -o $PWFILE -r $RECIPIENT
```

/cc @kholia do you have any suggestions / recommendations around something like this?

pfcoperez replied:
I was thinking about a ciphered PGP private key:

1. We generate the key-pair which ciphers and deciphers the payload from the user.
2. We  provide the public key from (1) to the customer.
3. We share the private key from (1) among Elastic peers using PGP packaged as Keybase messages or PGP messages ciphered using our PGP identities.

These key sharing steps are all internal, the customer only has to use a public PGP key that we provide.

The internal process of two-step payload deciphering can be automated in an internal tool. I think it would be quite easy to do so using PGP libraries for Go. This tiny tool could also serve as an assistant to handle the private key for (1) from Elastic engineer to Elastic engineer.

Finally, as this tool development is 100% independent from the changes in this script (this PR). I suggest going forward with it and follow up the work with the internal tool/script I've just described.

Wdyt @gcmurphy ?

gcmurphy replied:
Sure that works.

pfcoperez commented on diagnostics.sh:
```diff
@@ -196,6 +206,69 @@ get_docker(){
+        public_key_file=$1
+        target_file=$2
+        
+        temp_keyring=/tmp/$(date +%s%N)_pgp
```
++

pfcoperez replied:
Addressed in https://github.com/elastic/ece-support-diagnostics/pull/40/commits/622dfa0852e3f1a379b46382e15601dc03cf2e7f

pfcoperez replied:
Addressed in a2b04bd055e164513e966d128d56332cb1dee584

pfcoperez commented:
@s-nel I've addressed your comments, including the changes in the `README.md`, would you mind taking a look at the new doc entry? fe68069529ec7ce0611560b4bd171fff37027689

pfcoperez commented:
## QA Validation: Smoke test

In a `ce-aws`  test environment instance using the CI image for the PR: `2.5.0-SNAPSHOT_PR-42339`

### Grab ZK contents with exclusions and starting from a sub-tree

```bash
./diagnostics.sh -zk ./pablo.key.pub -zk-excluded '/container_sets/cloud-uis,/container_sets/zookeeper-servers' -zk-path '/container_sets' -d
```

```
Fri Nov 22 15:42:49 UTC 2019 [INFO]:  ECE Diagnostics
[2019-11-22 15:42:54,361][INFO ][no.found.zookeeper.settings.ConnectionCircuitBreakerSettings$] Loaded connection circuit breaker settings [RandomizedRetryForever(1 second,Some(1500 milliseconds))] {}
[2019-11-22 15:42:54,479][INFO ][org.apache.curator.utils.Compatibility] Using emulated InjectSessionExpiration {}
[2019-11-22 15:42:54,690][INFO ][org.apache.curator.framework.imps.CuratorFrameworkImpl] Starting {}
[2019-11-22 15:42:54,722][INFO ][no.found.curator.ForwardedEnsembleProvider] Resolved connection string from [http://containerhost:2180/zookeeper/clients/ensemble/connection-string?namespace=/v1] to [containerhost:22191/v1] with local namespace [/v1] {}
[2019-11-22 15:42:54,725][INFO ][org.apache.zookeeper.ZooKeeper] Initiating client connection, connectString=containerhost:22191/v1 sessionTimeout=15000 watcher=org.apache.curator.ConnectionState@425d5d46 {}
[2019-11-22 15:42:54,741][INFO ][org.apache.zookeeper.ClientCnxnSocket] jute.maxbuffer value is 4194304 Bytes {}
[2019-11-22 15:42:54,759][INFO ][org.apache.zookeeper.ClientCnxn] Opening socket connection to server containerhost/192.168.44.10:22191. Will not attempt to authenticate using SASL (unknown error) {}
[2019-11-22 15:42:54,772][INFO ][org.apache.zookeeper.ClientCnxn] Socket connection established, initiating session, client: /172.17.0.1:45380, server: containerhost/192.168.44.10:22191 {}
[2019-11-22 15:42:54,782][INFO ][org.apache.zookeeper.ClientCnxn] Session establishment complete on server containerhost/192.168.44.10:22191, sessionid = 0xa000001ced1002e, negotiated timeout = 15000 {}
[2019-11-22 15:42:54,785][INFO ][no.found.curator.ForwardedEnsembleProvider] Resolved connection string from [http://containerhost:2180/zookeeper/clients/ensemble/connection-string?namespace=/v1] to [containerhost:22191/v1] with local namespace [/v1] {}
[2019-11-22 15:42:54,790][INFO ][no.found.curator.ForwardedEnsembleProvider] Resolved connection string from [http://containerhost:2180/zookeeper/clients/ensemble/connection-string?namespace=/v1] to [containerhost:22191/v1] with local namespace [/v1] {}
[2019-11-22 15:42:54,797][INFO ][org.apache.curator.framework.state.ConnectionStateManager] State change: CONNECTED {}
[2019-11-22 15:42:54,798][INFO ][org.apache.curator.framework.imps.CuratorFrameworkImpl] Curator Schemas:

 {}
[2019-11-22 15:42:54,799][INFO ][no.found.shell.ShellBootstrap] Loading default ACLs from /secrets {}
[2019-11-22 15:42:54,809][INFO ][org.apache.curator.framework.imps.EnsembleTracker] New config event received: {version=100000000, server.10=localhost:2898:3898:participant;0.0.0.0:2191} {}
[2019-11-22 15:42:54,814][INFO ][org.apache.curator.framework.imps.EnsembleTracker] New config event received: {version=100000000, server.10=localhost:2898:3898:participant;0.0.0.0:2191} {}
[2019-11-22 15:42:54,836][INFO ][no.found.shell.ShellBootstrap] Loaded default ACLs from /secrets. {}
[2019-11-22 15:42:55,139][INFO ][akka.event.slf4j.Slf4jLogger] Slf4jLogger started {}
[2019-11-22 15:42:55,333][WARN ][no.found.curator.TreeCacheBuilder$] Using Curator TreeCache to watch ZK trees {}
[2019-11-22 15:42:57,152][INFO ][no.found.featureflags.staterepository.ZookeeperStateRepository] Feature flag updated: [GoProxyDeploy] with [FeatureState(true,None,None)] {}
[2019-11-22 15:42:57,156][INFO ][no.found.featureflags.staterepository.ZookeeperStateRepository] Feature flag updated: [GoProxyEnableHttpTraffic] with [FeatureState(true,None,None)] {}
[2019-11-22 15:42:57,157][INFO ][no.found.featureflags.staterepository.ZookeeperStateRepository] Feature flag updated: [HaProxyAllTraffic] with [FeatureState(true,None,None)] {}
[2019-11-22 15:42:57,159][INFO ][no.found.featureflags.staterepository.ZookeeperStateRepository] Feature flag updated: [ProxyPublishV2Metrics] with [FeatureState(true,None,None)] {}
[2019-11-22 15:42:57,164][INFO ][no.found.featureflags.staterepository.ZookeeperStateRepository] Feature flag updated: [ProxyWriteV2RoutingTable] with [FeatureState(true,None,None)] {}
[2019-11-22 15:42:57,165][INFO ][no.found.featureflags.staterepository.ZookeeperStateRepository] Feature flag updated: [StrictDeserialization] with [FeatureState(true,None,None)] {}
[2019-11-22 15:42:57,167][INFO ][no.found.featureflags.staterepository.ZookeeperStateRepository] Initialized cache for feature flag listener {}
Started Zookeeper contents dump into [/target/zkdump.zip] from ZK tree [zk:(/container_sets)] Some( excluding subtrees: [zk:(/container_sets/cloud-uis),zk:(/container_sets/zookeeper-servers)])
[2019-11-22 15:42:58,066][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/directors/shared {}
[2019-11-22 15:42:58,081][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/directors/containers {}
[2019-11-22 15:42:58,086][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/directors/events {}
[2019-11-22 15:42:58,089][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/allocators/shared {}
[2019-11-22 15:42:58,091][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/allocators/containers {}
[2019-11-22 15:42:58,097][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/allocators/events {}
[2019-11-22 15:42:58,099][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/runners/shared {}
[2019-11-22 15:42:58,101][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/runners/containers {}
[2019-11-22 15:42:58,104][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/runners/events {}
[2019-11-22 15:42:58,105][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/client-forwarders/shared {}
[2019-11-22 15:42:58,107][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/client-forwarders/containers {}
[2019-11-22 15:42:58,110][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/client-forwarders/events {}
[2019-11-22 15:42:58,112][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/constructors/shared {}
[2019-11-22 15:42:58,114][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/constructors/containers {}
[2019-11-22 15:42:58,123][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/constructors/events {}
[2019-11-22 15:42:58,127][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/blueprints/shared {}
[2019-11-22 15:42:58,129][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/blueprints/containers {}
[2019-11-22 15:42:58,134][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/blueprints/events {}
[2019-11-22 15:42:58,139][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/services-forwarders/shared {}
[2019-11-22 15:42:58,140][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/services-forwarders/containers {}
[2019-11-22 15:42:58,143][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/services-forwarders/events {}
[2019-11-22 15:42:58,145][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/allocator-metricbeats/shared {}
[2019-11-22 15:42:58,147][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/allocator-metricbeats/containers {}
[2019-11-22 15:42:58,150][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/allocator-metricbeats/events {}
[2019-11-22 15:42:58,152][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/proxies/shared {}
[2019-11-22 15:42:58,153][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/proxies/containers {}
[2019-11-22 15:42:58,158][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/proxies/events {}
[2019-11-22 15:42:58,160][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/beats-runners/shared {}
[2019-11-22 15:42:58,161][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/beats-runners/containers {}
[2019-11-22 15:42:58,164][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/beats-runners/events {}
[2019-11-22 15:42:58,166][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/curators/shared {}
[2019-11-22 15:42:58,167][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/curators/containers {}
[2019-11-22 15:42:58,170][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/curators/events {}
[2019-11-22 15:42:58,171][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/admin-consoles/shared {}
[2019-11-22 15:42:58,173][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/admin-consoles/containers {}
[2019-11-22 15:42:58,176][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/admin-consoles/events {}
[2019-11-22 15:42:58,177][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/client-observers/shared {}
[2019-11-22 15:42:58,179][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/client-observers/containers {}
[2019-11-22 15:42:58,182][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/client-observers/events {}
[2019-11-22 15:42:58,190][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/directors/events/slain {}
[2019-11-22 15:42:58,194][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/allocators/events/slain {}
[2019-11-22 15:42:58,198][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/runners/events/slain {}
[2019-11-22 15:42:58,209][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/client-forwarders/events/slain {}
[2019-11-22 15:42:58,212][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/constructors/events/slain {}
[2019-11-22 15:42:58,218][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/blueprints/events/slain {}
[2019-11-22 15:42:58,224][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/services-forwarders/events/slain {}
[2019-11-22 15:42:58,229][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/allocator-metricbeats/events/slain {}
[2019-11-22 15:42:58,237][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/proxies/events/slain {}
[2019-11-22 15:42:58,241][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/beats-runners/events/slain {}
[2019-11-22 15:42:58,245][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/curators/events/slain {}
[2019-11-22 15:42:58,248][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/admin-consoles/events/slain {}
[2019-11-22 15:42:58,252][INFO ][TreeTools                ] Found a node with non-JSON contents: /container_sets/client-observers/events/slain {}
Dump completed, output size = [95404] bytes
[2019-11-22 15:42:58,419][INFO ][org.apache.curator.framework.imps.CuratorFrameworkImpl] backgroundOperationsLoop exiting {}
[2019-11-22 15:42:58,433][INFO ][org.apache.zookeeper.ZooKeeper] Session: 0xa000001ced1002e closed {}
[2019-11-22 15:42:58,434][INFO ][org.apache.zookeeper.ClientCnxn] EventThread shut down for session: 0xa000001ced1002e {}
[2019-11-22 15:42:58,444][INFO ][no.found.util.LogApplicationExit$] Application is exiting {}
gpg: keybox '/tmp/tmp.McaXoUS9Lq/pubring.kbx' created
gpg: /tmp/tmp.McaXoUS9Lq/trustdb.gpg: trustdb created
gpg: key 7DD53647: public key "Pablo Francisco Pérez Hidalgo <pablo.perez@elastic.co>" imported
gpg: Total number processed: 1
gpg:               imported: 1
Fri Nov 22 15:43:00 UTC 2019 [INFO]:  Grabbing docker logs...
Fri Nov 22 15:43:00 UTC 2019 [INFO]:  Grabbing logs for containerId [7e44e35e6d1f]
Fri Nov 22 15:43:00 UTC 2019 [INFO]:  Grabbing logs for containerId [326353e6a476]
Fri Nov 22 15:43:01 UTC 2019 [INFO]:  Grabbing logs for containerId [d7a036b8ea4a]
Fri Nov 22 15:43:01 UTC 2019 [INFO]:  Grabbing logs for containerId [7ac72158d58b]
Fri Nov 22 15:43:01 UTC 2019 [INFO]:  Grabbing logs for containerId [a4e2e3b12af2]
Fri Nov 22 15:43:01 UTC 2019 [INFO]:  Grabbing logs for containerId [c05eff023e7e]
Fri Nov 22 15:43:01 UTC 2019 [INFO]:  Grabbing logs for containerId [107fbaf1fbda]
Fri Nov 22 15:43:01 UTC 2019 [INFO]:  Grabbing logs for containerId [abe3f0fa1db4]
Fri Nov 22 15:43:01 UTC 2019 [INFO]:  Grabbing logs for containerId [f60f087cd854]
Fri Nov 22 15:43:01 UTC 2019 [INFO]:  Grabbing logs for containerId [28d9f7f4faf1]
Fri Nov 22 15:43:01 UTC 2019 [INFO]:  Grabbing logs for containerId [e13fc39b7ba0]
Fri Nov 22 15:43:01 UTC 2019 [INFO]:  Grabbing logs for containerId [d966c7b2ee24]
Fri Nov 22 15:43:01 UTC 2019 [INFO]:  Grabbing logs for containerId [8490af4a0171]
Fri Nov 22 15:43:01 UTC 2019 [INFO]:  Grabbing logs for containerId [a813a102bbfe]
Fri Nov 22 15:43:01 UTC 2019 [INFO]:  Grabbing logs for containerId [09bf3d880f79]
Fri Nov 22 15:43:01 UTC 2019 [INFO]:  Grabbing logs for containerId [3147c7aca0e6]
Fri Nov 22 15:43:01 UTC 2019 [INFO]:  Grabbing logs for containerId [370f91dcc7ce]
Fri Nov 22 15:43:01 UTC 2019 [INFO]:  Grabbing logs for containerId [8a308465d8e1]
Fri Nov 22 15:43:01 UTC 2019 [INFO]:  Grabbing logs for containerId [5b84c5fe27ea]
Fri Nov 22 15:43:01 UTC 2019 [INFO]:  Grabbing logs for containerId [615b12357eec]
Fri Nov 22 15:43:01 UTC 2019 [INFO]:  Grabbing docker ps...
Fri Nov 22 15:43:02 UTC 2019 [INFO]:  Grabbing docker info...
Fri Nov 22 15:43:02 UTC 2019 [INFO]:  Grabbing docker images...
Fri Nov 22 15:43:02 UTC 2019 [INFO]:  Grabbing 5 repeated container stats...
Fri Nov 22 15:43:02 UTC 2019 [INFO]:  Grabbing docker stats 5
Fri Nov 22 15:43:04 UTC 2019 [INFO]:  Grabbing docker stats 4
Fri Nov 22 15:43:06 UTC 2019 [INFO]:  Grabbing docker stats 3
Fri Nov 22 15:43:08 UTC 2019 [INFO]:  Grabbing docker stats 2
Fri Nov 22 15:43:10 UTC 2019 [INFO]:  Grabbing docker stats 1
Fri Nov 22 15:43:12 UTC 2019 [INFO]:  Compressing diag file...
Fri Nov 22 15:43:12 UTC 2019 [INFO]:  Diag ready at /tmp/ece_diag_ip-192-168-44-10_22_Nov_2019_15_42_49.tar.gz
Fri Nov 22 15:43:12 UTC 2019 [INFO]:  Cleaning temp files...
```

On my local machine:

```bash
scp elastic@192.168.44.10:/tmp/ece_diag_ip-192-168-44-10_22_Nov_2019_15_42_49.tar.gz ./
tar xfz ece_diag_ip-192-168-44-10_22_Nov_2019_15_42_49.tar.gz

cd ece_diag_ip-192-168-44-10_22_Nov_2019_15_42_49/elastic/zookeeper_dump
keybase pgp decrypt -i ./zkdump.zip.gpg -o zkdump.zip
```

The file contains only contents under `/container_sets`:

```bash
unzip -l zkdump.zip
```

**Result** :ok: 

```
Archive:  zkdump.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
      290  11-22-2019 15:42   /container_sets/stats.json
        3  11-22-2019 15:42   /container_sets/contents.json
      289  11-22-2019 15:42   /container_sets/directors/stats.json
       23  11-22-2019 15:42   /container_sets/directors/contents.json
      289  11-22-2019 15:42   /container_sets/allocators/stats.json
       23  11-22-2019 15:42   /container_sets/allocators/contents.json
      289  11-22-2019 15:42   /container_sets/runners/stats.json
       23  11-22-2019 15:42   /container_sets/runners/contents.json
      289  11-22-2019 15:42   /container_sets/client-forwarders/stats.json
       23  11-22-2019 15:42   /container_sets/client-forwarders/contents.json
      289  11-22-2019 15:42   /container_sets/constructors/stats.json
       23  11-22-2019 15:42   /container_sets/constructors/contents.json
      289  11-22-2019 15:42   /container_sets/blueprints/stats.json
       23  11-22-2019 15:42   /container_sets/blueprints/contents.json
      289  11-22-2019 15:42   /container_sets/services-forwarders/stats.json
       23  11-22-2019 15:42   /container_sets/services-forwarders/contents.json
      289  11-22-2019 15:42   /container_sets/allocator-metricbeats/stats.json
       23  11-22-2019 15:42   /container_sets/allocator-metricbeats/contents.json
      289  11-22-2019 15:42   /container_sets/proxies/stats.json
       23  11-22-2019 15:42   /container_sets/proxies/contents.json
      289  11-22-2019 15:42   /container_sets/beats-runners/stats.json
       23  11-22-2019 15:42   /container_sets/beats-runners/contents.json
      289  11-22-2019 15:42   /container_sets/curators/stats.json
       23  11-22-2019 15:42   /container_sets/curators/contents.json
      289  11-22-2019 15:42   /container_sets/admin-consoles/stats.json
       23  11-22-2019 15:42   /container_sets/admin-consoles/contents.json
      289  11-22-2019 15:42   /container_sets/client-observers/stats.json
       23  11-22-2019 15:42   /container_sets/client-observers/contents.json
      289  11-22-2019 15:42   /container_sets/directors/shared/stats.json
       10  11-22-2019 15:42   /container_sets/directors/shared/contents.txt
      289  11-22-2019 15:42   /container_sets/directors/containers/stats.json
       10  11-22-2019 15:42   /container_sets/directors/containers/contents.txt
      290  11-22-2019 15:42   /container_sets/directors/secrets/stats.json
      408  11-22-2019 15:42   /container_sets/directors/secrets/contents.json
      289  11-22-2019 15:42   /container_sets/directors/events/stats.json
       10  11-22-2019 15:42   /container_sets/directors/events/contents.txt
      289  11-22-2019 15:42   /container_sets/allocators/shared/stats.json
       10  11-22-2019 15:42   /container_sets/allocators/shared/contents.txt
      289  11-22-2019 15:42   /container_sets/allocators/containers/stats.json
       10  11-22-2019 15:42   /container_sets/allocators/containers/contents.txt
      290  11-22-2019 15:42   /container_sets/allocators/secrets/stats.json
      409  11-22-2019 15:42   /container_sets/allocators/secrets/contents.json
      289  11-22-2019 15:42   /container_sets/allocators/events/stats.json
       10  11-22-2019 15:42   /container_sets/allocators/events/contents.txt
      289  11-22-2019 15:42   /container_sets/runners/shared/stats.json
       10  11-22-2019 15:42   /container_sets/runners/shared/contents.txt
      289  11-22-2019 15:42   /container_sets/runners/containers/stats.json
       10  11-22-2019 15:42   /container_sets/runners/containers/contents.txt
      290  11-22-2019 15:42   /container_sets/runners/secrets/stats.json
      406  11-22-2019 15:42   /container_sets/runners/secrets/contents.json
      289  11-22-2019 15:42   /container_sets/runners/events/stats.json
       10  11-22-2019 15:42   /container_sets/runners/events/contents.txt
      289  11-22-2019 15:42   /container_sets/client-forwarders/shared/stats.json
       10  11-22-2019 15:42   /container_sets/client-forwarders/shared/contents.txt
      289  11-22-2019 15:42   /container_sets/client-forwarders/containers/stats.json
       10  11-22-2019 15:42   /container_sets/client-forwarders/containers/contents.txt
      290  11-22-2019 15:42   /container_sets/client-forwarders/secrets/stats.json
      416  11-22-2019 15:42   /container_sets/client-forwarders/secrets/contents.json
      289  11-22-2019 15:42   /container_sets/client-forwarders/events/stats.json
       10  11-22-2019 15:42   /container_sets/client-forwarders/events/contents.txt
      289  11-22-2019 15:42   /container_sets/constructors/shared/stats.json
       10  11-22-2019 15:42   /container_sets/constructors/shared/contents.txt
      289  11-22-2019 15:42   /container_sets/constructors/containers/stats.json
       10  11-22-2019 15:42   /container_sets/constructors/containers/contents.txt
      290  11-22-2019 15:42   /container_sets/constructors/secrets/stats.json
      411  11-22-2019 15:42   /container_sets/constructors/secrets/contents.json
      289  11-22-2019 15:42   /container_sets/constructors/events/stats.json
       10  11-22-2019 15:42   /container_sets/constructors/events/contents.txt
      289  11-22-2019 15:42   /container_sets/blueprints/shared/stats.json
       10  11-22-2019 15:42   /container_sets/blueprints/shared/contents.txt
      289  11-22-2019 15:42   /container_sets/blueprints/containers/stats.json
       10  11-22-2019 15:42   /container_sets/blueprints/containers/contents.txt
      290  11-22-2019 15:42   /container_sets/blueprints/secrets/stats.json
      409  11-22-2019 15:42   /container_sets/blueprints/secrets/contents.json
      289  11-22-2019 15:42   /container_sets/blueprints/events/stats.json
       10  11-22-2019 15:42   /container_sets/blueprints/events/contents.txt
      289  11-22-2019 15:42   /container_sets/services-forwarders/shared/stats.json
       10  11-22-2019 15:42   /container_sets/services-forwarders/shared/contents.txt
      289  11-22-2019 15:42   /container_sets/services-forwarders/containers/stats.json
       10  11-22-2019 15:42   /container_sets/services-forwarders/containers/contents.txt
      290  11-22-2019 15:42   /container_sets/services-forwarders/secrets/stats.json
      418  11-22-2019 15:42   /container_sets/services-forwarders/secrets/contents.json
      289  11-22-2019 15:42   /container_sets/services-forwarders/events/stats.json
       10  11-22-2019 15:42   /container_sets/services-forwarders/events/contents.txt
      289  11-22-2019 15:42   /container_sets/allocator-metricbeats/shared/stats.json
       10  11-22-2019 15:42   /container_sets/allocator-metricbeats/shared/contents.txt
      289  11-22-2019 15:42   /container_sets/allocator-metricbeats/containers/stats.json
       10  11-22-2019 15:42   /container_sets/allocator-metricbeats/containers/contents.txt
      290  11-22-2019 15:42   /container_sets/allocator-metricbeats/secrets/stats.json
      420  11-22-2019 15:42   /container_sets/allocator-metricbeats/secrets/contents.json
      289  11-22-2019 15:42   /container_sets/allocator-metricbeats/events/stats.json
       10  11-22-2019 15:42   /container_sets/allocator-metricbeats/events/contents.txt
      289  11-22-2019 15:42   /container_sets/proxies/shared/stats.json
       10  11-22-2019 15:42   /container_sets/proxies/shared/contents.txt
      289  11-22-2019 15:42   /container_sets/proxies/containers/stats.json
       10  11-22-2019 15:42   /container_sets/proxies/containers/contents.txt
      290  11-22-2019 15:42   /container_sets/proxies/secrets/stats.json
      406  11-22-2019 15:42   /container_sets/proxies/secrets/contents.json
      289  11-22-2019 15:42   /container_sets/proxies/events/stats.json
       10  11-22-2019 15:42   /container_sets/proxies/events/contents.txt
      289  11-22-2019 15:42   /container_sets/beats-runners/shared/stats.json
       10  11-22-2019 15:42   /container_sets/beats-runners/shared/contents.txt
      289  11-22-2019 15:42   /container_sets/beats-runners/containers/stats.json
       10  11-22-2019 15:42   /container_sets/beats-runners/containers/contents.txt
      290  11-22-2019 15:42   /container_sets/beats-runners/secrets/stats.json
      412  11-22-2019 15:42   /container_sets/beats-runners/secrets/contents.json
      289  11-22-2019 15:42   /container_sets/beats-runners/events/stats.json
       10  11-22-2019 15:42   /container_sets/beats-runners/events/contents.txt
      289  11-22-2019 15:42   /container_sets/curators/shared/stats.json
       10  11-22-2019 15:42   /container_sets/curators/shared/contents.txt
      289  11-22-2019 15:42   /container_sets/curators/containers/stats.json
       10  11-22-2019 15:42   /container_sets/curators/containers/contents.txt
      290  11-22-2019 15:42   /container_sets/curators/secrets/stats.json
      407  11-22-2019 15:42   /container_sets/curators/secrets/contents.json
      289  11-22-2019 15:42   /container_sets/curators/events/stats.json
       10  11-22-2019 15:42   /container_sets/curators/events/contents.txt
      289  11-22-2019 15:42   /container_sets/admin-consoles/shared/stats.json
       10  11-22-2019 15:42   /container_sets/admin-consoles/shared/contents.txt
      289  11-22-2019 15:42   /container_sets/admin-consoles/containers/stats.json
       10  11-22-2019 15:42   /container_sets/admin-consoles/containers/contents.txt
      290  11-22-2019 15:42   /container_sets/admin-consoles/secrets/stats.json
      413  11-22-2019 15:42   /container_sets/admin-consoles/secrets/contents.json
      289  11-22-2019 15:42   /container_sets/admin-consoles/events/stats.json
       10  11-22-2019 15:42   /container_sets/admin-consoles/events/contents.txt
      289  11-22-2019 15:42   /container_sets/client-observers/shared/stats.json
       10  11-22-2019 15:42   /container_sets/client-observers/shared/contents.txt
      289  11-22-2019 15:42   /container_sets/client-observers/containers/stats.json
       10  11-22-2019 15:42   /container_sets/client-observers/containers/contents.txt
      290  11-22-2019 15:42   /container_sets/client-observers/secrets/stats.json
      415  11-22-2019 15:42   /container_sets/client-observers/secrets/contents.json
      289  11-22-2019 15:42   /container_sets/client-observers/events/stats.json
       10  11-22-2019 15:42   /container_sets/client-observers/events/contents.txt
      291  11-22-2019 15:42   /container_sets/directors/containers/director/stats.json
     1892  11-22-2019 15:42   /container_sets/directors/containers/director/contents.json
      289  11-22-2019 15:42   /container_sets/directors/events/slain/stats.json
       10  11-22-2019 15:42   /container_sets/directors/events/slain/contents.txt
      291  11-22-2019 15:42   /container_sets/allocators/containers/allocator/stats.json
     3120  11-22-2019 15:42   /container_sets/allocators/containers/allocator/contents.json
      289  11-22-2019 15:42   /container_sets/allocators/events/slain/stats.json
       10  11-22-2019 15:42   /container_sets/allocators/events/slain/contents.txt
      291  11-22-2019 15:42   /container_sets/runners/containers/runner/stats.json
     2884  11-22-2019 15:42   /container_sets/runners/containers/runner/contents.json
      289  11-22-2019 15:42   /container_sets/runners/events/slain/stats.json
       10  11-22-2019 15:42   /container_sets/runners/events/slain/contents.txt
      291  11-22-2019 15:42   /container_sets/client-forwarders/containers/client-forwarder/stats.json
     1300  11-22-2019 15:42   /container_sets/client-forwarders/containers/client-forwarder/contents.json
      289  11-22-2019 15:42   /container_sets/client-forwarders/events/slain/stats.json
       10  11-22-2019 15:42   /container_sets/client-forwarders/events/slain/contents.txt
      291  11-22-2019 15:42   /container_sets/constructors/containers/constructor/stats.json
     2042  11-22-2019 15:42   /container_sets/constructors/containers/constructor/contents.json
      289  11-22-2019 15:42   /container_sets/constructors/events/slain/stats.json
       10  11-22-2019 15:42   /container_sets/constructors/events/slain/contents.txt
      291  11-22-2019 15:42   /container_sets/blueprints/containers/blueprint/stats.json
     1356  11-22-2019 15:42   /container_sets/blueprints/containers/blueprint/contents.json
      289  11-22-2019 15:42   /container_sets/blueprints/events/slain/stats.json
       10  11-22-2019 15:42   /container_sets/blueprints/events/slain/contents.txt
      291  11-22-2019 15:42   /container_sets/services-forwarders/containers/services-forwarder/stats.json
     2109  11-22-2019 15:42   /container_sets/services-forwarders/containers/services-forwarder/contents.json
      289  11-22-2019 15:42   /container_sets/services-forwarders/events/slain/stats.json
       10  11-22-2019 15:42   /container_sets/services-forwarders/events/slain/contents.txt
      291  11-22-2019 15:42   /container_sets/allocator-metricbeats/containers/allocator-metricbeat/stats.json
     1384  11-22-2019 15:42   /container_sets/allocator-metricbeats/containers/allocator-metricbeat/contents.json
      289  11-22-2019 15:42   /container_sets/allocator-metricbeats/events/slain/stats.json
       10  11-22-2019 15:42   /container_sets/allocator-metricbeats/events/slain/contents.txt
      291  11-22-2019 15:42   /container_sets/proxies/containers/proxyv2/stats.json
     2287  11-22-2019 15:42   /container_sets/proxies/containers/proxyv2/contents.json
      291  11-22-2019 15:42   /container_sets/proxies/containers/route-server/stats.json
     1564  11-22-2019 15:42   /container_sets/proxies/containers/route-server/contents.json
      289  11-22-2019 15:42   /container_sets/proxies/events/slain/stats.json
       10  11-22-2019 15:42   /container_sets/proxies/events/slain/contents.txt
      291  11-22-2019 15:42   /container_sets/beats-runners/containers/beats-runner/stats.json
     2216  11-22-2019 15:42   /container_sets/beats-runners/containers/beats-runner/contents.json
      289  11-22-2019 15:42   /container_sets/beats-runners/events/slain/stats.json
       10  11-22-2019 15:42   /container_sets/beats-runners/events/slain/contents.txt
      291  11-22-2019 15:42   /container_sets/curators/containers/curator/stats.json
     1472  11-22-2019 15:42   /container_sets/curators/containers/curator/contents.json
      289  11-22-2019 15:42   /container_sets/curators/events/slain/stats.json
       10  11-22-2019 15:42   /container_sets/curators/events/slain/contents.txt
      291  11-22-2019 15:42   /container_sets/admin-consoles/containers/admin-console/stats.json
     3293  11-22-2019 15:42   /container_sets/admin-consoles/containers/admin-console/contents.json
      289  11-22-2019 15:42   /container_sets/admin-consoles/events/slain/stats.json
       10  11-22-2019 15:42   /container_sets/admin-consoles/events/slain/contents.txt
      291  11-22-2019 15:42   /container_sets/client-observers/containers/client-observer/stats.json
     1750  11-22-2019 15:42   /container_sets/client-observers/containers/client-observer/contents.json
      289  11-22-2019 15:42   /container_sets/client-observers/events/slain/stats.json
       10  11-22-2019 15:42   /container_sets/client-observers/events/slain/contents.txt
      306  11-22-2019 15:42   /container_sets/directors/containers/director/allocation@192.168.44.10/stats.json
       21  11-22-2019 15:42   /container_sets/directors/containers/director/allocation@192.168.44.10/contents.json
      308  11-22-2019 15:42   /container_sets/directors/containers/director/inspect@192.168.44.10/stats.json
     9514  11-22-2019 15:42   /container_sets/directors/containers/director/inspect@192.168.44.10/contents.json
      306  11-22-2019 15:42   /container_sets/allocators/containers/allocator/allocation@192.168.44.10/stats.json
       21  11-22-2019 15:42   /container_sets/allocators/containers/allocator/allocation@192.168.44.10/contents.json
      308  11-22-2019 15:42   /container_sets/allocators/containers/allocator/inspect@192.168.44.10/stats.json
    11392  11-22-2019 15:42   /container_sets/allocators/containers/allocator/inspect@192.168.44.10/contents.json
      306  11-22-2019 15:42   /container_sets/runners/containers/runner/allocation@192.168.44.10/stats.json
       21  11-22-2019 15:42   /container_sets/runners/containers/runner/allocation@192.168.44.10/contents.json
      308  11-22-2019 15:42   /container_sets/runners/containers/runner/inspect@192.168.44.10/stats.json
    11222  11-22-2019 15:42   /container_sets/runners/containers/runner/inspect@192.168.44.10/contents.json
      306  11-22-2019 15:42   /container_sets/client-forwarders/containers/client-forwarder/allocation@192.168.44.10/stats.json
       21  11-22-2019 15:42   /container_sets/client-forwarders/containers/client-forwarder/allocation@192.168.44.10/contents.json
      308  11-22-2019 15:42   /container_sets/client-forwarders/containers/client-forwarder/inspect@192.168.44.10/stats.json
     7482  11-22-2019 15:42   /container_sets/client-forwarders/containers/client-forwarder/inspect@192.168.44.10/contents.json
      306  11-22-2019 15:42   /container_sets/constructors/containers/constructor/allocation@192.168.44.10/stats.json
       21  11-22-2019 15:42   /container_sets/constructors/containers/constructor/allocation@192.168.44.10/contents.json
      308  11-22-2019 15:42   /container_sets/constructors/containers/constructor/inspect@192.168.44.10/stats.json
     9631  11-22-2019 15:42   /container_sets/constructors/containers/constructor/inspect@192.168.44.10/contents.json
      306  11-22-2019 15:42   /container_sets/blueprints/containers/blueprint/allocation@192.168.44.10/stats.json
       21  11-22-2019 15:42   /container_sets/blueprints/containers/blueprint/allocation@192.168.44.10/contents.json
      308  11-22-2019 15:42   /container_sets/blueprints/containers/blueprint/inspect@192.168.44.10/stats.json
     8473  11-22-2019 15:42   /container_sets/blueprints/containers/blueprint/inspect@192.168.44.10/contents.json
      306  11-22-2019 15:42   /container_sets/services-forwarders/containers/services-forwarder/allocation@192.168.44.10/stats.json
       21  11-22-2019 15:42   /container_sets/services-forwarders/containers/services-forwarder/allocation@192.168.44.10/contents.json
      308  11-22-2019 15:42   /container_sets/services-forwarders/containers/services-forwarder/inspect@192.168.44.10/stats.json
     9932  11-22-2019 15:42   /container_sets/services-forwarders/containers/services-forwarder/inspect@192.168.44.10/contents.json
      306  11-22-2019 15:42   /container_sets/allocator-metricbeats/containers/allocator-metricbeat/allocation@192.168.44.10/stats.json
       21  11-22-2019 15:42   /container_sets/allocator-metricbeats/containers/allocator-metricbeat/allocation@192.168.44.10/contents.json
      308  11-22-2019 15:42   /container_sets/allocator-metricbeats/containers/allocator-metricbeat/inspect@192.168.44.10/stats.json
     8416  11-22-2019 15:42   /container_sets/allocator-metricbeats/containers/allocator-metricbeat/inspect@192.168.44.10/contents.json
      306  11-22-2019 15:42   /container_sets/proxies/containers/proxyv2/allocation@192.168.44.10/stats.json
       21  11-22-2019 15:42   /container_sets/proxies/containers/proxyv2/allocation@192.168.44.10/contents.json
      308  11-22-2019 15:42   /container_sets/proxies/containers/proxyv2/inspect@192.168.44.10/stats.json
    10357  11-22-2019 15:42   /container_sets/proxies/containers/proxyv2/inspect@192.168.44.10/contents.json
      306  11-22-2019 15:42   /container_sets/proxies/containers/route-server/allocation@192.168.44.10/stats.json
       21  11-22-2019 15:42   /container_sets/proxies/containers/route-server/allocation@192.168.44.10/contents.json
      308  11-22-2019 15:42   /container_sets/proxies/containers/route-server/inspect@192.168.44.10/stats.json
     8813  11-22-2019 15:42   /container_sets/proxies/containers/route-server/inspect@192.168.44.10/contents.json
      306  11-22-2019 15:42   /container_sets/beats-runners/containers/beats-runner/allocation@192.168.44.10/stats.json
       21  11-22-2019 15:42   /container_sets/beats-runners/containers/beats-runner/allocation@192.168.44.10/contents.json
      308  11-22-2019 15:42   /container_sets/beats-runners/containers/beats-runner/inspect@192.168.44.10/stats.json
    10365  11-22-2019 15:42   /container_sets/beats-runners/containers/beats-runner/inspect@192.168.44.10/contents.json
      306  11-22-2019 15:42   /container_sets/curators/containers/curator/allocation@192.168.44.10/stats.json
       21  11-22-2019 15:42   /container_sets/curators/containers/curator/allocation@192.168.44.10/contents.json
      308  11-22-2019 15:42   /container_sets/curators/containers/curator/inspect@192.168.44.10/stats.json
     8600  11-22-2019 15:42   /container_sets/curators/containers/curator/inspect@192.168.44.10/contents.json
      306  11-22-2019 15:42   /container_sets/admin-consoles/containers/admin-console/allocation@192.168.44.10/stats.json
       21  11-22-2019 15:42   /container_sets/admin-consoles/containers/admin-console/allocation@192.168.44.10/contents.json
      308  11-22-2019 15:42   /container_sets/admin-consoles/containers/admin-console/inspect@192.168.44.10/stats.json
    10648  11-22-2019 15:42   /container_sets/admin-consoles/containers/admin-console/inspect@192.168.44.10/contents.json
---------                     -------
   194860                     238 files
```

But not '/container_sets/cloud-uis` or `/container_sets/zookeeper-servers':

```bash
unzip -l zkdump.zip | grep 'cloud-uis'
```

**Result** EMPTY :ok: 

```bash
unzip -l zkdump.zip | grep 'zookeeper-servers'
```

**Result** EMPTY :ok:

pfcoperez commented:
> Typo noted, otherwise LGTM

Thanks @s-nel ! I just used the Spanish word ;P

pfcoperez commented:
@s-nel I've added a check to let customers using old versions of ECE know that they need ECE 2.5+

If you approve, I'll merge.
